sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Helpers/CollectionUtils.cs

```diff
@@ -82,6 +82,46 @@ public static IEnumerable<T> WhereNotNull<T>(this IEnumerable<T> enumerable)
             where T : class
         {
             return enumerable.Where(e => e != null);
+        }
+
+        /// <summary>
+        /// Applies a specified function to the corresponding elements of two sequences,
+        /// producing a sequence of the results. If the collections have different length
+        /// null will be passed in the operation function for the corresponding items that 
+        /// do not exist.
+        /// </summary>
+        /// <typeparam name="TFirst">The type of the elements of the first input sequence.</typeparam>
+        /// <typeparam name="TSecond">The type of the elements of the second input sequence.</typeparam>
+        /// <typeparam name="TResult">The type of the elements of the result sequence.</typeparam>
+        /// <param name="first">The first sequence to merge.</param>
+        /// <param name="second">The second sequence to merge.</param>
+        /// <param name="operation">A function that specifies how to merge the elements from the two sequences.</param>
+        /// <returns>An System.Collections.Generic.IEnumerable`1 that contains merged elements of
+        /// two input sequences.</returns>
+        public static IEnumerable<TResult> Merge<TFirst, TSecond, TResult>(this IEnumerable<TFirst> first,

```

Comment:
You mean nest usings? I disagree on that. For me it's the same as writing if (a && b) or  if (a) { if (b) {}}. The first is shorter and captures the intent better - there is no hierarchy and order of iter1 and iter2 doesn't matter.